js/charts/stackedhorizontalbar.js

```diff
@@ -336,62 +358,97 @@ function stackedhorizontalbar() {
 
     var drawViz = function (element) {
         var me = this;
-
-        element.append('rect')
-            .style('fill', function (d, i) {
-                return UTIL.getDisplayColor(_measure.indexOf(d.key), _displayColor);
-            })
-            .style('stroke', function (d, i) {
-                return UTIL.getBorderColor(_measure.indexOf(d.key), _borderColor);
-            })
-            .attr("y", function (d) {
-                return x(d.data[_dimension[0]]);
-            })
-            .attr("x", function (d) {
-                return (d[0] < d[1]) ? (y(d[0]) + 1) : (y(d[1]) + 1);
-            })
-            .attr("width", function (d) {
-                return Math.abs(y(d[1]) - y(d[0]));
-            })
-            .attr("height", x.bandwidth())
-            .style('stroke-width', 2)
-            .on('mouseover', _handleMouseOverFn.call(chart, tooltip, _local_svg))
-            .on('mousemove', _handleMouseMoveFn.call(chart, tooltip, _local_svg))
-            .on('mouseout', _handleMouseOutFn.call(chart, tooltip, _local_svg))
-            .on('click', function (d) {
-                if ($("#myonoffswitch").prop('checked') == false) {
-                    $('#Modal_' + $(div).attr('id') + ' .measure').val(d.key);
-                    $('#Modal_' + $(div).attr('id') + ' .threshold').val('');
-                    $('#Modal_' + $(div).attr('id') + ' .measure').attr('disabled', true);;
-                    $('#Modal_' + $(div).attr('id')).modal('toggle');
-                }
-                else {
-                    var confirm = d3.select(div).select('.confirm')
-                        .style('visibility', 'visible');
-                    var _filter = _Local_data.filter(function (d1) {
-                        return d.data[_dimension[0]] === d1[_dimension[0]]
-                    })
-                    var rect = d3.select(this);
-                    if (rect.classed('selected')) {
-                        rect.classed('selected', false);
-                        filterData.map(function (val, i) {
-                            if (val[_dimension[0]] == d[_dimension[0]]) {
-                                filterData.splice(i, 1)
-                            }
+        if (!_print) {
+            element.append('rect')
```

Comment:
We can keep the element rendering part outside the if block and only keep the mouse events part inside it